instrumentation/opentelemetry-instrumentation-logging/src/opentelemetry/instrumentation/logging/__init__.py

```diff
@@ -0,0 +1,115 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=no-value-for-parameter,no-member,no-name-in-module
+
+import logging  # pylint: disable=import-self
+from os import environ
+
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.logging.constants import (
+    _MODULE_DOC,
+    DEFAULT_LOGGING_FORMAT,
+)
+from opentelemetry.instrumentation.logging.environment_variables import (
+    OTEL_PYTHON_LOG_CORRELATION,
+    OTEL_PYTHON_LOG_FORMAT,
+    OTEL_PYTHON_LOG_LEVEL,
+)
+from opentelemetry.trace import (
+    INVALID_SPAN,
+    INVALID_SPAN_CONTEXT,
+    get_current_span,
+    get_tracer_provider,
+)
+
+__doc__ = _MODULE_DOC
+
+LEVELS = {
+    "debug": logging.DEBUG,
+    "info": logging.INFO,
+    "warning": logging.WARNING,
+    "error": logging.ERROR,
+}
+
+
+class LoggingInstrumentor(BaseInstrumentor):  # pylint: disable=empty-docstring
+    __doc__ = """An instrumentor for stdlib logging module.
+
+    This instrumentor injects tracing context into logging records and optionally sets the global logging format to the following:
+
+    .. code-block::
+
+        {default_logging_format}
+
+    See `BaseInstrumentor`
+    """.format(
+        default_logging_format=DEFAULT_LOGGING_FORMAT
+    )
+
+    _factory_registered = False
+    _is_instrumented = False
+
+    def _instrument(self, **kwargs):
+        LoggingInstrumentor._is_instrumented = True
+        service_name = ""
+        provider = kwargs.get("tracer_provider", None) or get_tracer_provider()
+        resource = provider.resource if provider else None
+        if resource:
+            service_name = resource.attributes.get("service.name")
+
+        old_factory = logging.getLogRecordFactory()
+        if not LoggingInstrumentor._factory_registered:
+            LoggingInstrumentor._factory_registered = True
+
+            def record_factory(*args, **kwargs):
+                record = old_factory(*args, **kwargs)
+                if not LoggingInstrumentor._is_instrumented:
+                    return record
+
+                record.otelSpanID = ""
+                record.otelTraceID = ""
```

Comment:
If it is invalid span, it will log `trace_id= span_id=`? Would leaving the zeroes be better i.e. `trace_id=0 span_id=0`?